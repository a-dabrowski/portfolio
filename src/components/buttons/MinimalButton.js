import React from "react";

import "./MinimalButton.scss";
const Cta = ({isFull, children, isFixedWidth}) => {
  return (
    <button className={`minimal-button ${isFixedWidth && 'sm:w-72'}`}>
      {children}
    </button>
  );
};


export default Cta
