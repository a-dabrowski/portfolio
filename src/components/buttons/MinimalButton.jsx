import React from 'react';

const MinimalButton = ({
  isFull,
  children,
  isFixedWidth,
  className,
  disabled,
}) => {
  return (
    <button
      disabled={disabled || false}
      className={`px-6 py-4 uppercase transition bg-white text-black hover:bg-black hover:text-white border-solid border-2 border-black disabled:bg-gray-300 disabled:text-gray-600 disabled:cursor-not-allowed ${
        isFixedWidth ? 'sm:w-72' : ''
      } ${className || ''}`}
    >
      {children}
    </button>
  );
};

export { MinimalButton };
export default MinimalButton;
