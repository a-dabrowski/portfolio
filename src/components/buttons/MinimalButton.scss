.minimal-button {
  border: 2px solid black;
  background: white;
  color: black;
  transition: all .3s;
  padding: 24px;
  text-transform: uppercase;
  &:hover {
    color: white;
    background: black;
  }
}
