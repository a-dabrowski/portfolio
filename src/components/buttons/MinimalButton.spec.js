import React from 'react';
import { render } from '@testing-library/react';

import MinimalButton from './MinimalButton';

describe('MinimalButton component', () => {
  it('renders properly', () => {
    const { container, getByText, asFragment } = render(
      <MinimalButton>Button test</MinimalButton>
    );
    expect(
      asFragment(<MinimalButton>Button Test</MinimalButton>)
    ).toMatchSnapshot();
  });
  it('show the same text as declared', () => {
    const { getByText } = render(<MinimalButton>Button test</MinimalButton>);
    expect(getByText('Button test')).toBeInTheDocument();
  });
});
