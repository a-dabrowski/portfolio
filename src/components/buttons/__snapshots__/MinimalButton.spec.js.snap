// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalButton component renders properly 1`] = `
<DocumentFragment>
  <button
    class="p-6 uppercase transition bg-white text-black hover:bg-black hover:text-white border-solid border-2 border-black undefined"
  >
    Button test
  </button>
</DocumentFragment>
`;
